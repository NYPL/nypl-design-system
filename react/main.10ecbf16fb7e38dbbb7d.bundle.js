(window.webpackJsonp=window.webpackJsonp||[]).push([[0],{1292:function(n,o,c){"use strict";c.r(o);c(1293),c(1323),c(1335),c(1442),c(1485)},1448:function(n,o){},1449:function(n,o){},1450:function(n,o){},1451:function(n,o){},501:function(n,o,c){c(502),c(645),n.exports=c(1292)},564:function(n,o){}},[[501,1,2]]]);