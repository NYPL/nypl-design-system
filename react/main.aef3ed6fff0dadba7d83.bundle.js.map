{"version":3,"file":"main.aef3ed6fff0dadba7d83.bundle.js","sources":["webpack:///main.aef3ed6fff0dadba7d83.bundle.js"],"mappings":"AAAA","sourceRoot":""}