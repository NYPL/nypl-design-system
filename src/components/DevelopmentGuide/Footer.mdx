import { Meta, Source } from "@storybook/blocks";

<Meta title="Development Guide/Footer" />

# Footer

## Table of Contents

- [Overview](#overview)
- [Implementation](#implementation)
- [Accessibility](#accessibility)

## Overview

This `Footer` component renders the NYPL-branded footer elements such as
navigational NYPL.org links, social media links, copyright, and NYPL building
facade image. This component is expected to be rendered at the end of a webpage,
before the closing `</body>` tag. This component lives on the NYPL `Header` app.

- [NYPL Footer Preview](https://ds-header.nypl.org/footer)
- [NYPL Header App Github repo](https://github.com/NYPL/nypl-header-app)

## Implementation

The `Footer` can be embedded on an NYPL app through the following code snippet:

<Source
  code={`
<div id="nypl-footer"></div>
<script type="module" src="https://ds-header.nypl.org/footer.min.js?containerId=nypl-footer" async></script>
`}
  language="html"
/>

## Accessibility

The `Footer` component renders as an HTML `<footer>` element with a WAI-ARIA role
of `role="contentinfo"`. This is the common practice for the footer landmark
region on a page to maximize its accessibility in browsers and assistive
technologies.

Special consideration should be made regarding the placement of the `Footer`
component in the DOM. Rendering this component, or similar HTML landmark region
components, inside another landmark region causes accessibility issues. For
example, rendering the `Footer` component inside an `aside` HTML element is
an error.

Resources:

- [W3C WAI Page Structure Code Example](https://www.w3.org/WAI/tutorials/page-structure/example/)
- [W3C WAI Page Regions](https://www.w3.org/WAI/tutorials/page-structure/regions/)
