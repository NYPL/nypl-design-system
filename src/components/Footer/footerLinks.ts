const link = {
  nyplLinks: [
    [
      {
        text: "Accessibility",
        href: "http://www.nypl.org/accessibility",
      },
      {
        text: "Press",
        href: "http://www.nypl.org/help/about-nypl/media-center",
      },
      {
        text: "Careers",
        href: "http://www.nypl.org/careers",
      },
      {
        text: "Space Rental",
        href: "http://www.nypl.org/spacerental",
      },
    ],
    [
      {
        text: "Privacy Policy",
        href: "http://www.nypl.org/help/about-nypl/legal-notices/privacy-policy",
      },
      {
        text: "Other Policies",
        href: "http://www.nypl.org/policies",
      },
      {
        text: "Terms & Conditions",
        href: "http://www.nypl.org/terms-conditions",
      },
      {
        text: "Governance",
        href: "http://www.nypl.org/help/about-nypl/leadership/board-trustees",
      },
    ],
    [
      {
        text: "Rules & Regulations",
        href: "http://www.nypl.org/help/about-nypl/legal-notices/rules-and-regulations",
      },
      {
        text: "About NYPL",
        href: "http://www.nypl.org/help/about-nypl",
      },
      {
        text: "Language",
        href: "http://www.nypl.org/language",
      },
    ],
  ],
  socialMedia: [
    {
      href: "https://www.facebook.com/nypl",
      iconName: "legacySocialFacebook",
      title: "NYPL on Facebook",
    },
    {
      href: "https://twitter.com/nypl",
      iconName: "legacySocialTwitter",
      title: "NYPL on Twitter",
    },
    {
      href: "https://instagram.com/nypl",
      iconName: "legacySocialInstagram",
      title: "NYPL on Instagram",
    },
    {
      href: "https://www.youtube.com/user/NewYorkPublicLibrary",
      iconName: "legacySocialYoutube",
      title: "NYPL on Youtube",
    },
  ],
};

export default link;
