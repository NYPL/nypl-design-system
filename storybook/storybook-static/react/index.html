<!doctype html><html lang="en"><head><meta charset="utf-8"><title>Storybook</title><meta name="viewport" content="width=device-width,initial-scale=1"><style>html, body {
    overflow: hidden;
    height: 100%;
    width: 100%;
    margin: 0;
    padding: 0;
  }</style><script>/* globals window */
  /* eslint-disable no-underscore-dangle */
  try {
    if (window.top !== window) {
      window.__REACT_DEVTOOLS_GLOBAL_HOOK__ = window.top.__REACT_DEVTOOLS_GLOBAL_HOOK__;
    }
  } catch (e) {
    // eslint-disable-next-line no-console
    console.warn('unable to connect to top frame for connecting dev tools');
  }</script><style>#root[hidden],
      #docs-root[hidden] {
        display: none !important;
      }</style></head><body><div id="root"></div><div id="docs-root"></div><script>window['DOCS_MODE'] = false;</script><script src="runtime~main.00bd9d033d7a0b972fc9.bundle.js"></script><script src="vendors~main.d2b6aafef1a259d165c0.bundle.js"></script><script src="main.3bf71672a6dd35499004.bundle.js"></script></body></html>