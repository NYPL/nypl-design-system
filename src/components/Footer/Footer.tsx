import { Box, chakra, useMultiStyleConfig } from "@chakra-ui/react";
import React, { forwardRef } from "react";

import Icon, { IconNames } from "../Icons/Icon";
import Image from "../Image/Image";
import Link from "../Link/Link";
import List from "../List/List";
import Logo from "../Logo/Logo";
import linkData from "./footerLinks";

interface FooterProps {
  /** Additional class name to add. */
  className?: string;
  /** ID that other components can cross reference for accessibility purposes */
  id?: string;
}

/**
 * This `Footer` component renders the NYPL-branded footer elements such
 * as navigational NYPL.org links, social media links, copyright, and
 * NYPL building facade image.
 */
export const Footer = chakra(
  forwardRef<HTMLDivElement, FooterProps>(
    ({ className, id = "footer", ...rest }, ref?) => {
      const styles = useMultiStyleConfig("Footer", {});

      const nyplLinks = linkData.nyplLinks.map((links, index) => (
        <List
          key={index}
          listItems={links.map((link) => (
            <Link href={link.href} key={link.text}>
              {link.text}
            </Link>
          ))}
          noStyling
          type="ul"
          __css={styles.footerLinksInner}
        />
      ));
      const socialLinks = linkData.socialMedia.map((link, index) => (
        <Link href={link.href} key={index}>
          <Icon
            decorative={false}
            name={link.iconName as IconNames}
            size="small"
            title={link.title}
          />
        </Link>
      ));

      return (
        <Box
          as="footer"
          className={className}
          id={id}
          ref={ref}
          role="contentinfo"
          {...rest}
          __css={styles}
        >
          <Box __css={styles.listsContainer}>
            <List
              data-testid="text-links"
              listItems={nyplLinks}
              noStyling
              type="ul"
              __css={styles.footerLinksList}
            />
            <List
              data-testid="social-media-links"
              listItems={socialLinks}
              noStyling
              type="ul"
              __css={styles.socialMediaList}
            />
          </Box>
          <Box __css={styles.nyplInfoContainer}>
            <Box>
              <Image
                alt="NYPL Main Building Facade"
                src="https://cdn-d8.nypl.org/s3fs-public/2020-05/NYPL_MainFacadeRev2Cam2.png"
              />
            </Box>
            <Box __css={styles.copyright}>
              <p>Â© The New York Public Library, {new Date().getFullYear()}</p>
              <p>The New York Public Library is a 501(c)(3) | EIN 13-1887440</p>
            </Box>
          </Box>
          <Box __css={styles.logoContainer}>
            <Logo
              decorative={false}
              name="nyplTextWhite"
              size="small"
              title="The New York Public Library"
            />
          </Box>
        </Box>
      );
    }
  )
);

export default Footer;
