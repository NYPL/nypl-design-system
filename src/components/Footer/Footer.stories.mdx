import {
  ArgsTable,
  Canvas,
  Description,
  Meta,
  Story,
} from "@storybook/addon-docs";
import { withDesign } from "storybook-addon-designs";

import Footer from "./Footer";
import { getCategory } from "../../utils/componentCategories";
import DSProvider from "../../theme/provider";

<Meta
  title={getCategory("Footer")}
  component={Footer}
  decorators={[withDesign]}
  parameters={{
    jest: ["Footer.test.tsx"],
  }}
  argTypes={{
    className: { control: false },
    id: { control: false },
  }}
/>

# Footer

| Component Version | DS Version |
| ----------------- | ---------- |
| Added             | `1.1.0`    |
| Latest            | `1.1.2`    |

## Table of Contents

- [Overview](#overview)
- [Component Props](#component-props)
- [Accessibility](#accessibility)
- [With DS Template Components](#with-ds-templates)

## Overview

<Description of={Footer} />

This component is expected to be rendered at the end of a webpage, before the
closing `</body>` tag. There are no expected props for this component so it can
be rendered simply as `<Footer />`.

```tsx
<body>
  <header>...</header>
  <main>...</main>

  <!-- Add the Footer component here -->
  <Footer />
</body>
```

## Component Props

<Canvas withToolbar>
  <Story name="Footer">
    <Footer />
  </Story>
</Canvas>

Note: There are no custom props for the `Footer` component.

## Accessibility

The `Footer` component renders as an HTML `<footer>` element with a WAI-ARIA role
of `role="contentinfo"`. This is the common practice for the footer landmark
region on a page to maximize its accessibility in browsers and assistive
technologies.

Special consideration should be made regarding the placement of the `Footer`
component in the DOM. Rendering this component, or similar HTML landmark region
components, inside another landmark region causes accessibility issues. For
example, rendering the `Footer` component inside an `aside` HTML element is
an error.

Resources:

- [W3C WAI Page Structure Code Example](https://www.w3.org/WAI/tutorials/page-structure/example/)
- [W3C WAI Page Regions](https://www.w3.org/WAI/tutorials/page-structure/regions/)

## With DS Template Components

The `Footer` component is a page layout component and it can be used within the
DS `Template`s components.

### TemplateAppContainer Component

The DS' `TemplateAppContainer` component renders an HTML `<footer>` element when
content is passed to its `footer` prop. If your application uses the
`TemplateAppContainer` component and the `Footer` component, make sure to also
set the `renderFooterElement` prop to `false`. Both components will render HTML
`<footer>` elements and we only need one; setting `renderFooterElement` to
`false` will prevent rendering nested `<footer>` elements.

```tsx
import {
  Footer,
  TemplateAppContainer,
} from "@nypl/design-system-react-components";

const MyApp => (
  <TemplateAppContainer
    header={...}
    contentPrimary={...}
    contentSidebar={...}
    sidebar="left"
    footer={<Footer />}
    renderFooterElement={false}
  />
);
```

If this step is not done, the `TemplateAppContainer` component will log a warning
to the browser's console.

### Individual Template Components

When using individual DS `Template` components, the `TemplateFooter` component
should be used to render the `<footer>` HTML element. When using with the DS
`Footer` component, set the `renderFooterElement` prop to `false` to prevent
rendering nested `<footer>` HTML elements.

```tsx
import {
  Footer,
  Template,
  TemplateHeader,
  TemplateContent,
  TemplateContentPrimary,
  TemplateContentSidebar,
  TemplateFooter,
} from "@nypl/design-system-react-components";

const MyApp => (
  <Template>
    <TemplateHeader> ... </TemplateHeader>
    <TemplateContent sidebar="left">
      <TemplateContentPrimary> ... </TemplateContentPrimary>
    </TemplateContent>
    <TemplateFooter renderFooterElement={false}>
    <!-- Add the Footer component here -->
      <Footer />
    </TemplateFooter>
</Template>
);
```

If this step is not done, the `TemplateFooter` component will log a warning to
the browser's console.
