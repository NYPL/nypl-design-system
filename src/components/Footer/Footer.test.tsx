import { render, screen, within } from "@testing-library/react";
import { axe } from "jest-axe";
import * as React from "react";
import { mockAllIsIntersecting } from "react-intersection-observer/test-utils";
import renderer from "react-test-renderer";

import Footer from "./Footer";

describe("Footer Accessibility", () => {
  it("passes axe accessibility", async () => {
    // Mock IntersectionObserver to render images.
    mockAllIsIntersecting(true);

    const { container } = render(<Footer />);
    expect(await axe(container)).toHaveNoViolations();
  });
});

describe("Footer", () => {
  beforeEach(() => {
    render(<Footer />);
    // Mock IntersectionObserver to render images.
    mockAllIsIntersecting(true);
  });

  it("renders a footer element container", () => {
    expect(screen.getByRole("contentinfo")).toBeInTheDocument();
  });

  it("renders a list of three link lists", () => {
    const list = screen.getByTestId("text-links");
    const innerLists = within(list).getAllByRole("list");

    expect(innerLists).toHaveLength(3);
    expect(within(innerLists[0]).getAllByRole("link")).toHaveLength(4);
    expect(within(innerLists[0]).getAllByRole("link")[0]).toHaveTextContent(
      "Accessibility"
    );
    expect(within(innerLists[1]).getAllByRole("link")).toHaveLength(4);
    expect(within(innerLists[1]).getAllByRole("link")[0]).toHaveTextContent(
      "Privacy Policy"
    );
    expect(within(innerLists[2]).getAllByRole("link")).toHaveLength(3);
    expect(within(innerLists[2]).getAllByRole("link")[0]).toHaveTextContent(
      "Rules & Regulations"
    );
  });

  it("renders a list of four social media icons", () => {
    const list = screen.getByTestId("social-media-links");
    const icons = within(list).getAllByRole("img");

    expect(icons).toHaveLength(4);
    expect(icons[0]).toHaveAttribute("title", "NYPL on Facebook");
    expect(icons[1]).toHaveAttribute("title", "NYPL on Twitter");
    expect(icons[2]).toHaveAttribute("title", "NYPL on Instagram");
    expect(icons[3]).toHaveAttribute("title", "NYPL on Youtube");
  });

  it("renders a footer image of the NYPL building", () => {
    const logo = screen.getAllByRole("img")[4];

    expect(logo).toBeInTheDocument();
    expect(logo).toHaveAttribute(
      "src",
      "https://cdn-d8.nypl.org/s3fs-public/2020-05/NYPL_MainFacadeRev2Cam2.png"
    );
  });

  it("renders the NYPL logo", () => {
    expect(
      screen.getByTitle("The New York Public Library")
    ).toBeInTheDocument();
  });

  it("Renders the UI snapshot correctly", () => {
    const footer = renderer.create(<Footer />).toJSON();
    expect(footer).toMatchSnapshot();
  });
});
