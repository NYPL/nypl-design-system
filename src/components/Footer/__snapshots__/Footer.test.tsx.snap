// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders the UI snapshot correctly 1`] = `
<footer
  className="css-1xdhyk6"
  id="footer"
  role="contentinfo"
>
  <div
    className="css-0"
  >
    <ul
      className="css-1xdhyk6"
      data-testid="text-links"
    >
      <li>
        <ul
          className="css-1xdhyk6"
        >
          <li>
            <a
              className="css-1xdhyk6"
              href="http://www.nypl.org/accessibility"
              rel={null}
              target={null}
            >
              Accessibility
            </a>
          </li>
          <li>
            <a
              className="css-1xdhyk6"
              href="http://www.nypl.org/help/about-nypl/media-center"
              rel={null}
              target={null}
            >
              Press
            </a>
          </li>
          <li>
            <a
              className="css-1xdhyk6"
              href="http://www.nypl.org/careers"
              rel={null}
              target={null}
            >
              Careers
            </a>
          </li>
          <li>
            <a
              className="css-1xdhyk6"
              href="http://www.nypl.org/spacerental"
              rel={null}
              target={null}
            >
              Space Rental
            </a>
          </li>
        </ul>
      </li>
      <li>
        <ul
          className="css-1xdhyk6"
        >
          <li>
            <a
              className="css-1xdhyk6"
              href="http://www.nypl.org/help/about-nypl/legal-notices/privacy-policy"
              rel={null}
              target={null}
            >
              Privacy Policy
            </a>
          </li>
          <li>
            <a
              className="css-1xdhyk6"
              href="http://www.nypl.org/policies"
              rel={null}
              target={null}
            >
              Other Policies
            </a>
          </li>
          <li>
            <a
              className="css-1xdhyk6"
              href="http://www.nypl.org/terms-conditions"
              rel={null}
              target={null}
            >
              Terms & Conditions
            </a>
          </li>
          <li>
            <a
              className="css-1xdhyk6"
              href="http://www.nypl.org/help/about-nypl/leadership/board-trustees"
              rel={null}
              target={null}
            >
              Governance
            </a>
          </li>
        </ul>
      </li>
      <li>
        <ul
          className="css-1xdhyk6"
        >
          <li>
            <a
              className="css-1xdhyk6"
              href="http://www.nypl.org/help/about-nypl/legal-notices/rules-and-regulations"
              rel={null}
              target={null}
            >
              Rules & Regulations
            </a>
          </li>
          <li>
            <a
              className="css-1xdhyk6"
              href="http://www.nypl.org/help/about-nypl"
              rel={null}
              target={null}
            >
              About NYPL
            </a>
          </li>
          <li>
            <a
              className="css-1xdhyk6"
              href="http://www.nypl.org/language"
              rel={null}
              target={null}
            >
              Language
            </a>
          </li>
        </ul>
      </li>
    </ul>
    <ul
      className="css-1xdhyk6"
      data-testid="social-media-links"
    >
      <li>
        <a
          className="css-1xdhyk6"
          href="https://www.facebook.com/nypl"
          rel={null}
          target={null}
        >
          <svg
            aria-hidden={false}
            className="chakra-icon css-1grhd2q"
            focusable={false}
            role="img"
            title="NYPL on Facebook"
            viewBox="0 0 24 24"
          >
            <g
              stroke="currentColor"
              strokeWidth="1.5"
            >
              <path
                d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
                fill="none"
                strokeLinecap="round"
              />
              <path
                d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
                fill="currentColor"
                strokeLinecap="round"
              />
              <circle
                cx="12"
                cy="12"
                fill="none"
                r="11.25"
                strokeMiterlimit="10"
              />
            </g>
          </svg>
        </a>
      </li>
      <li>
        <a
          className="css-1xdhyk6"
          href="https://twitter.com/nypl"
          rel={null}
          target={null}
        >
          <svg
            aria-hidden={false}
            className="chakra-icon css-1grhd2q"
            focusable={false}
            role="img"
            title="NYPL on Twitter"
            viewBox="0 0 24 24"
          >
            <g
              stroke="currentColor"
              strokeWidth="1.5"
            >
              <path
                d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
                fill="none"
                strokeLinecap="round"
              />
              <path
                d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
                fill="currentColor"
                strokeLinecap="round"
              />
              <circle
                cx="12"
                cy="12"
                fill="none"
                r="11.25"
                strokeMiterlimit="10"
              />
            </g>
          </svg>
        </a>
      </li>
      <li>
        <a
          className="css-1xdhyk6"
          href="https://instagram.com/nypl"
          rel={null}
          target={null}
        >
          <svg
            aria-hidden={false}
            className="chakra-icon css-1grhd2q"
            focusable={false}
            role="img"
            title="NYPL on Instagram"
            viewBox="0 0 24 24"
          >
            <g
              stroke="currentColor"
              strokeWidth="1.5"
            >
              <path
                d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
                fill="none"
                strokeLinecap="round"
              />
              <path
                d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
                fill="currentColor"
                strokeLinecap="round"
              />
              <circle
                cx="12"
                cy="12"
                fill="none"
                r="11.25"
                strokeMiterlimit="10"
              />
            </g>
          </svg>
        </a>
      </li>
      <li>
        <a
          className="css-1xdhyk6"
          href="https://www.youtube.com/user/NewYorkPublicLibrary"
          rel={null}
          target={null}
        >
          <svg
            aria-hidden={false}
            className="chakra-icon css-1grhd2q"
            focusable={false}
            role="img"
            title="NYPL on Youtube"
            viewBox="0 0 24 24"
          >
            <g
              stroke="currentColor"
              strokeWidth="1.5"
            >
              <path
                d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
                fill="none"
                strokeLinecap="round"
              />
              <path
                d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
                fill="currentColor"
                strokeLinecap="round"
              />
              <circle
                cx="12"
                cy="12"
                fill="none"
                r="11.25"
                strokeMiterlimit="10"
              />
            </g>
          </svg>
        </a>
      </li>
    </ul>
  </div>
  <div
    className="css-0"
  >
    <div
      className="css-0"
    >
      <div
        className="css-0"
        data-testid="test"
      >
        <img
          alt="NYPL Main Building Facade"
          className="css-0"
          src="https://cdn-d8.nypl.org/s3fs-public/2020-05/NYPL_MainFacadeRev2Cam2.png"
        />
      </div>
    </div>
    <div
      className="css-0"
    >
      <p>
        Â© The New York Public Library, 
        2022
      </p>
      <p>
        The New York Public Library is a 501(c)(3) | EIN 13-1887440
      </p>
    </div>
  </div>
  <div
    className="css-0"
  >
    <svg
      aria-hidden={false}
      className="chakra-icon css-1grhd2q"
      focusable={false}
      role="img"
      title="The New York Public Library"
      viewBox="0 0 24 24"
    >
      <g
        stroke="currentColor"
        strokeWidth="1.5"
      >
        <path
          d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
          fill="none"
          strokeLinecap="round"
        />
        <path
          d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
          fill="currentColor"
          strokeLinecap="round"
        />
        <circle
          cx="12"
          cy="12"
          fill="none"
          r="11.25"
          strokeMiterlimit="10"
        />
      </g>
    </svg>
  </div>
</footer>
`;
